### PR TITLE
Add request saving feature

### PR DESCRIPTION
**Pull request by steffenix**
The idea is to be able to make requests from using this endpoint and being able to get those requests afterward something like [requestb.in](http://requestb.in).

My use case would be to test API that require API keys that changes overtime (like APNS and GCM) and being able to make sure that the request has been done with the correct parameters.

### CODE REVIEW

**Comment by edgurgel** [comment]
It's an interesting feature but  I would propose some changes:

Change `/store` and `retreive` to be `/store/:id` so that multiple users can send different ids and retrieve different ids

Then we can have 1 ETS table holding all the requests and each tuple is 

`{ id, request }`

And they can have more than 1 with `id`

Then if `retrieve/id` is requested all these tuples are removed, aggregated and sent.

We should also have some way to TTL the tuples of an id when they get too old. So that the ETS table does not consume too much memory?

What do you think?

**Comment by steffenix** [comment]
The idea of having an id for each user is great, I din't think about it because I use this for local testing but it's definitely a feature needed for the heroku hosted version.

Would you be ok if we add `ecache`as a dependency to resolve the TTL part?

**Comment by edgurgel** [comment]
What about https://github.com/sasa1977/con_cache ? I never used ecache but if you have experience maybe it's good enough?

**Comment by steffenix** [comment]
Can't get why this test does not pass, would you give me an hint @edgurgel

File: test/retrieve_request_handler_tests.exs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+defmodule HTTParrot.RetrieveRequestHandlerTests do
+  use ExUnit.Case
+  import :meck
+  import HTTParrot.RetrieveRequestHandler
+
+  setup do
+    HTTParrot.RequestStore.clear(:test)
```
Comment (edgurgel) [inline test/retrieve_request_handler_tests.exs]: Because we are using meck, you need to add this to the setup:

`on_exit fn -> unload end`

**Comment by edgurgel** [comment]
Cool after pulling your branch I noticed that we test on travis using:

`mix test --no-start` (.travis.yml)

We need to remove this `--no-start` then con_cache will be up an running 👍

File: lib/httparrot/request_store.ex [hunk 2]
```diff
@@ -0,0 +1,20 @@
+defmodule HTTParrot.RequestStore do
+
```
Comment (edgurgel) [inline lib/httparrot/request_store.ex]: I wonder if you could add a `@moduledoc` to the module, @doc and @spec to the functions?

Comment (edgurgel) [inline lib/httparrot/request_store.ex]: Shall we also add tests just for the request store?

Comment (steffenix) [inline lib/httparrot/request_store.ex]: We could be we could also think that this is tested by the HTTPs tests and not write any as you wish.

**Comment by steffenix** [comment]
I have added the doc fixed the test and added test on the RequestStore.
I had to update the elixir 1.2 version because of a bug with OTP 19 support.
https://github.com/elixir-lang/elixir/issues/4617

Please review.

**Comment by edgurgel** [comment]
Thanks @steffenix! I will release a new version soon! 🎉

**Pull request merged**